contracts/handlers/fee/FeeHandlerGeneric.sol

```diff
@@ -0,0 +1,92 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.11;
+
+import "./FeeHandlerWithOracle.sol";
+
+/**
+    @title Handles deposit fees for generic messages based on Effective rates provided by Fee oracle.
+    @author ChainSafe Systems.
+    @notice This contract is intended to be used with the Bridge contract.
+ */
+contract FeeHandlerGeneric is FeeHandlerWithOracle {
+
+    /**
+        @param bridgeAddress Contract address of previously deployed Bridge.
+        @param feeHandlerRouterAddress Contract address of previously deployed FeeHandlerRouter.
+     */
+    constructor(address bridgeAddress, address feeHandlerRouterAddress) FeeHandlerWithOracle(bridgeAddress, feeHandlerRouterAddress) {
+    }
+    
+     /**
+        @notice Calculates fee for generic messages.
+        This function is almost identical to the _calculateFee function in the base contract.
+        The difference is the txCost calculation formula.
+        @param sender Sender of the deposit.
+        @param fromDomainID ID of the source chain.
+        @param destinationDomainID ID of chain deposit will be bridged to.
+        @param resourceID ResourceID to be used when making deposits.
+        @param depositData Additional data to be passed to specified handler.
+        @param feeData Additional data to be passed to the fee handler.
+        @return fee Returns the fee amount.
+        @return tokenAddress Returns the address of the token to be used for fee.
+     */
+    function _calculateFee(address sender, uint8 fromDomainID, uint8 destinationDomainID, bytes32 resourceID, bytes calldata depositData, bytes calldata feeData) internal view override returns(uint256 fee, address tokenAddress) {
```

Comment:
Should the `feeData.amount` be compared against `depositData.amount` as anyone could set 0 in `feeData.amount` and pay the minimum amount instead of the percentage of amount of tokens actually sent?

Reply:
Or would it be easier to use depositData.amount instead of feeData.amount as it seems it is the same value anyway.

Reply:
Yes I gues some confusion happened in the initial implementation. depositData.amount should be used to calculate fee, and feeData.amount should be used as a maximum fee that user is willing to pay.

Reply:
Things could be clarified with renaming `feeData.amount` to `feeData.maxFeeAmount` or alike.

Reply:
wait, but feeData is signed by feeOracle and it can be changed by anyone.

Hm ,frankly speaking this:
> Yes I guess some confusion happened in the initial implementation. `depositData.amount` should be used to calculate fee, and `feeData.amount` should be used as a maximum fee that user is willing to pay.
is not true. I mean we never was expecting user to set maxFee he willing to pay.

Generally idea was just to cover feecosts on dest network with this calculations.

Reply:
`feeData.amount` currently is not signed by the oracle, it's just appended to the signed message.

Reply:
It is not signed because oracle doesn't know about the amount of the transfer I guess.

Reply:
So what are we going to do with feeData.amount? @lastperson @P1sar @mpetrun5 
A fix to use depositData.amount is required anyway. How should we use feeData.amount?
1. Compare depositData.amount to feeData.amount and just revert if they differ 
or
2. Use feeData.amount as max fee and revert if it's exceeded? (I don't remember that we were going to use it as max fee and I cannot find the references for it but we can do it now)

Reply:
yes, so we should use depositData.amount for fee calculation to avoid this, and do some check with feeData.amount

Reply:
> So what are we going to do with feeData.amount? @lastperson @P1sar @mpetrun5 A fix to use depositData.amount is required anyway. How should we use feeData.amount?
> 
> 1. Compare depositData.amount to feeData.amount and just revert if they differ
>    or
> 2. Use feeData.amount as max fee and revert if it's exceeded? (I don't remember that we were going to use it as max fee and I cannot find the references for it but we can do it now)

It makes sense for me to add feeData.amount as maxFeeAmount and revert if the fee is higher.
It would also make more sense with the generic handler.

Also, should we add a 0 as an exception to not check the max fee amount?

Reply:
> Also, should we add a 0 as an exception to not check the max fee amount?

Sounds good to me.

Reply:
I like this idea that we use feeData.amount as maxFeeAmount in the context of generic messages - adds the ability for the user to safeguard additionally, also treating 0 as a special case to not check the max fee amount sounds good :+1: